CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+#  Contribution Guidelines and Policies
+
+## Contribution Guidelines
+
+We encourage contributions to the NetworKit source code.
+If you encounter any questions regarding NetworKit development open an issue on GitHub or contact the [mailing list][list].
+
+### General Guidelines
+
+- NetworKit comes with a large collection of unit tests.
+  Please provide tests for new code and for behavior that is modified.
+  See below for instructions on building and running the tests.
+- We generally welcome Jupyter notebooks that demonstrate how new algorithms are used. If you add a new feature,
+  consider adding a notebook to the `notebooks/` directory (or add your feature to an existing notebook).
+
+### Coding Style
+
+- We use 4 space indentation.
+- Adapt to the existing naming conventions. NetworKit generally uses lowerCamelCase
+  for (global and local) variables and UpperCamelCase for classes.
+- We have a `.clang-format` file in the root directory. We do not require all code to follow the coding style specified
+  in this file. However, it should be taken as a suggestion.
+
+###  Usage of Git
+
+We prefer a linear Git history. In particular:
+- Properly rebase your commits on the development branch before posting or updating a PR.
+- Do not merge from development branches into feature branches/PRs for no good reason.
+- Force push to the feature branches of your fork to update PRs.
+
+###  Building and Running Unit tests
+
+NetworKit comes with a large suite of unit tests. We use the Google Test framework to define and run our tests.
+The unit tests can only be run from a clone or copy of the repository and not from a pip installation. In order to run the unit tests, you need to compile them first.
+
+This is done by setting the [CMake] `NETWORKI_BUILD_TESTS` flag to `ON`:
+
+    cmake -DNETWORKIT_BUILD_TESTS=ON ..
+
+Single tests can be executed with:
+
+    ./networkit_tests --gtest_filter=CentralityGTest.testBetweennessCentrality
+
+Additionally, one can specify the level of the logs outputs by adding `--loglevel=<log_level>`;
+supported log levels are: `TRACE`, `DEBUG`, `INFO`, `WARN`, `ERROR`, and `FATAL`. For more details, see the
+output of `./networkit_tests -h`.
+
+## Policies and Maintainance
+
+### Supported Compilers
+
+We aim to support all releases of GCC and Clang for 5 years. In effect, this means that we can only use C++ features
+that have been available in a released GCC or Clang compiler for more than 5 years.
+
+### Deprecation and Removal of APIs
+
+Features need to be deprecated for two releases before they are removed.
+
+*Breaking API changes* should be kept to a minimum. If breaking API changes are done, they should be
```

Comment:
With the new release cycle, I think we guarantee a stable ABI across minor and patch releases? I would also suggest name "bugfix releases" "patch releases", since they not necessarily fix things and this would be in line with library-naming conventions.

Is CHANGES.md an official file used by github? Only found something about CHANGELOG.md.

Reply:
I don't know whether there is an official file name for GitHub.